packages/genesys-spark-components/src/components/beta/gux-time-zone/gux-time-zone.tsx

```diff
@@ -0,0 +1,72 @@
+import { Component, JSX, h, Host, Prop, Element } from '@stencil/core';
+import { trackComponent } from '@utils/tracking/usage';
+import { buildI18nForComponent, GetI18nValue } from '../../../i18n';
+import translationResources from '../gux-time-zone-picker/i18n/en.json';
+import { GuxTimeZoneIdentifier } from '../../../i18n/time-zone/types';
+
+import { getLocalizedOffset, shortenZone } from './gux-time-zone.service';
+
+@Component({
+  tag: 'gux-time-zone-beta',
+  shadow: true
+})
+export class GuxTimeZoneBeta {
+  private i18n: GetI18nValue;
+
+  /**
+   * Reference to the host element.
+   */
+  @Element()
+  root: HTMLElement;
+
+  /**
+   * The id of the time zone to display
+   */
+  @Prop()
+  timeZoneId: GuxTimeZoneIdentifier;
+
+  /**
+   * True to display the zone's offset from UTC
+   */
+  @Prop()
+  offset: boolean;
+
+  /**
+   * True to display surround the offset with parentheses
+   */
+  @Prop()
+  surroundOffset: boolean;
+
+  /**
+   * True to shorten the displayed zone name: 'Europe/London' -> 'London'
+   */
+  @Prop()
+  shorten: boolean;
+
+  private renderZoneDisplay(): JSX.Element | undefined {
+    let localizedZone = this.i18n(this.timeZoneId);
+    if (this.shorten) {
+      localizedZone = shortenZone(localizedZone);
+    }
+    let displayText = localizedZone;
+    if (this.offset) {
+      const localizedUTC = this.i18n('UTC');
+      const localizedOffset = getLocalizedOffset(localizedUTC, this.timeZoneId);
+
+      displayText = `${localizedZone} ${localizedOffset}`;
+      if (this.surroundOffset) {
+        displayText = `${localizedZone} (${localizedOffset})`;
+      }
+    }
+    return (<span>{displayText}</span>) as JSX.Element;
```

Comment:
I feel like the code would be more testable if instead this function returned `displayText` and then the consuming code rendered the surrounding span tag.